npm/vite-dev-server/src/resolveConfig.ts

```diff
@@ -122,6 +125,7 @@ function makeCypressViteConfig (config: ViteDevServerConfig, vite: Vite): Inline
           vite.searchForWorkspaceRoot?.(process.cwd()),
         ],
       },
+      port: vitePort,
```

Comment:
Could just be `port` - if `port` is provided, it's a number, otherwise it'll be undefined. No need for the `vitePort` variable